packages/core/src/__stampers__/passkey/base.ts

```diff
@@ -0,0 +1,165 @@
+            transports: cred.transports,
+          })) as any,
+        });
+      } catch (error) {
+        throw new Error(
+          `Failed to load passkey stamper for react-native: ${error}`,
+        );
+      }
+    } else {
+      throw new Error("Unsupported platform for passkey stamper");
+    }
+  }
+
+  async stamp(payload: string): Promise<TStamp> {
+    return await this.stamper.stamp(payload);
+  }
+
+  /**
+   * Create a passkey for an end-user, taking care of various lower-level details.
+   *
+   * @returns {Promise<Passkey>}
+   */
+  createWebPasskey = async (
+    config: Record<any, any> = {},
+  ): Promise<Passkey> => {
+    const challenge = generateRandomBuffer();
+    const encodedChallenge = base64StringToBase64UrlEncodedString(
+      btoa(String.fromCharCode(...new Uint8Array(challenge))),
+    );
+    const authenticatorUserId = generateRandomBuffer();
+
+    // WebAuthn credential options options can be found here:
+    // https://www.w3.org/TR/webauthn-2/#sctn-sample-registration
+    //
+    // All pubkey algorithms can be found here: https://www.iana.org/assignments/cose/cose.xhtml#algorithms
+    // Turnkey only supports ES256 (-7) and RS256 (-257)
+    //
+    // The pubkey type only supports one value, "public-key"
+    // See https://www.w3.org/TR/webauthn-2/#enumdef-publickeycredentialtype for more details
+    // TODO: consider un-nesting these config params
+    const webauthnConfig: CredentialCreationOptions = {
+      publicKey: {
+        rp: {
+          id: config.publicKey?.rp?.id ?? this.config.rpId,
+          name: config.publicKey?.rp?.name ?? "",
+        },
+        challenge: config.publicKey?.challenge ?? challenge,
+        pubKeyCredParams: config.publicKey?.pubKeyCredParams ?? [
+          {
+            type: "public-key",
+            alg: -7,
+          },
+          {
+            type: "public-key",
+            alg: -257,
+          },
+        ],
+        user: {
+          id: config.publicKey?.user?.id ?? authenticatorUserId,
+          name: config.publicKey?.user?.name ?? "Default User",
+          displayName: config.publicKey?.user?.displayName ?? "Default User",
+        },
+        authenticatorSelection: {
+          authenticatorAttachment:
+            config.publicKey?.authenticatorSelection?.authenticatorAttachment ??
+            undefined, // default to empty
+          requireResidentKey:
+            config.publicKey?.authenticatorSelection?.requireResidentKey ??
+            true,
+          residentKey:
+            config.publicKey?.authenticatorSelection?.residentKey ?? "required",
+          userVerification:
+            config.publicKey?.authenticatorSelection?.userVerification ??
+            "preferred",
+        },
+      },
+    };
+
+    const attestation = await getWebAuthnAttestation(webauthnConfig);
+
+    return {
+      encodedChallenge: config.publicKey?.challenge
+        ? base64StringToBase64UrlEncodedString(config.publicKey?.challenge)
```

Comment:
feel like if we're in this clause then we would expect `config.publicKey?` to be defined? so 

```suggestion
        ? base64StringToBase64UrlEncodedString(config.publicKey!.challenge)
```